### ISSUE TITLE
UC Folder Task List Bug (NN)

### ISSUE THREAD

**Issue by SurbhiBakshi** [cityofaustin/atd-data-tech#2511]
**As a ROW Permit Review Analyst reviewing my task list , we should not have to enter an End Date and time in the task manually to have the UC Folder permit task come off of the Reviewers Task List. The date should already be there ad we should only need to review and approve**

**Stakeholders:** Jacelyn, Isiah
**DTS Manager:** Nate

**Requirements:**

- Have the Initial Distribution Process End Date on their task list to automatically Populate and mirror the End Date and Time entered from the Process tab
- Reviewer should not have to enter the End Date and Time manually on UC Tasks
LICARIR: 
- Right-of-Way Admin 50075
- Permit Completeness Check 51065
- Initial Distribution process code 51070
- Permit Case Manager Review 51095
- Right-of-Way Review 51295
- Adjust Review Due Dates 61060

**Comment by SurbhiBakshi** [comment]
Email from Jaclyn L - 

I wanted to mention that for the UC folder, these two tasks show up after I process each UC application.

Permit Completeness Check, Initial Distribution

 

Is there a way to fix that for UC folders so I don’t have to go to each task and insert an arbitrary date and time?

**Comment by SurbhiBakshi** [comment]
Hi Jaclyn, 
Thanks for bringing this problem to my attention. I will create an issue in the DTS Amanda backlog. We might need more information. One of us will contact you for more details. 


Best, 
Surbhi

**Comment by TracyLinder** [comment]
User research needed in order to provide a sustainable solution. @Nadin-Nader will set something up with Jaclyn.

**Comment by Nadin-Nader** [comment]
Notes found from Meeting on 4/22/2020:
UC Permit Task List Issue Meeting with Jaclyn, Isaiah and Surbhi 4/22/2020:

Appears to be a process in the folder that is disconnected, these cases are not coming off of task list. Could also be a work Flow issue.

AMANDA USERID: LEWALLEN

Once any folder goes to active, approved and released, closed or approved, it should come off of task list and it does not. User hasnt been able to use the task list because of 1000s of items in the task list.

Noticed it 3 or 4 years ago but didnt become an issue til about 2 years ago when he needed to use the function.

User usually completes the task with the three process attempts but it doesnt come off. He mentione dthat he did not enter an Start and End date manually.

Jackie troubleshot it twice by trying to clear a task. She entered the Start and End date manually and it goes away. On the first try she received an error stating 'unable to Update process status contact the help desk' (thinking it may be permissions to the folder)

Tried it on ROW Review Closed UC Folder 

Most of the folders that are sill on but dont go away are:
UC FOLDER types stay and some RW
GP, SP go away like they are supposed to

Steps:
Tasklist> look up user> LEWALLEN (see recordning for other processes that might have the same issue.

Processes:
ROW Review
Permit case review manager (on this one Jaclyn entered start and end day and received no error and the task went away

They would like to know:
Is there a way to make the end date and start date not required to close the folder task???

Asked Isaiah to go through the tasklist and clear what he could by entering the date manually. We also asked that he filter/sort through as much to group them together and that he provide a date range of the ones that can be taken off. This will make clearing them easier for CTM and provide a couple of different ways to try to complete this, if possible.

He will send the list next week along with what he was able to do. We will then evaluate and get with CTM to see what can be done and find out if the Start and End date field has to be required or if it can be changed.
[https://app.zenhub.com/files/140626918/0dd53678-2478-48f1-971e-320e714488f5/download](https://app.zenhub.com/files/140626918/0dd53678-2478-48f1-971e-320e714488f5/download)

**Comment by Nadin-Nader** [comment]
My research notes so far (5/13/2020):

Want:They want to not have to enter a date to have the task purged from task list

Questions for Jaclyn:
-How do they use the folder/Process each UC Application?
Need Steps on how they use the folder
-What status does the folder come in and what does it change to when the Permit Completeness Check Process is inserted?
-Does the Initial Distribution Process get inserted with the Permit Completeness Check Process?
Or 
-Does it get inserted after the Permit Completeness Check Process?
-What status does the Initial Distribution Process change to when added or does it have no effect?
-When they are complete, what status do they have to be in/should be in to close? (if multiple)
-Does any other status close the task automatically or do they all have to closed manually via an arbitrary date?
-Are they dependent on each other?
-Do they have to be in the UC folder or can the folder on have one?
*The way to fix it is to have the procedure remove the folder from the User's task list by adding the End date automatically when the folder is set to a particular status on the Initial Distribution Process and/or the Permit Completeness Check Process

Or Make it to where the UC folder is not dependent on an End date to come off of the Task list

My UAT:
Applied for permit (Franchise Utility 50685), Added block, No Excavation RW, Im the contact/Applicant, 2020-003269 UC; 
Row ID 12388036
After Submittal i had no fees to pay.
Folder was in a "Internet Pending" Status
Applicant was Me
Info Tab: Distribution (Ace Chilled Water Review), ACE Review, AW Infrastructure Management Review, Electric Review, GAATAN Review, Street and Bridge Review, Traffic Control Review, Traffic Signal Review, Watershed Engineering Review was set to Yes
-No Fees
Process Tab had one Process "Web Application Acceptance" and the Assigned User was "ROW INTAKE GROUP"
Attempts for Process are:
Accepted, Change Permit, Department Comments, Plans Received, Rejected

My UAT 2:
Applied for Permit (Franchise Utility 50685), Added a block (1000 Barton Springs Road), Excavation in ROW = yes, Im the contact/Applicant, 2020-003271 UC

Scheduling User Research with Jaclyn

**Comment by Nadin-Nader** [comment]
From: Nader, Nadin 
Sent: Thursday, May 14, 2020 8:41 AM
To: Lozano, Jaclyn <Jaclyn.Lozano@austintexas.gov>
Subject: #2511 UC Folder/ Task List Bug and Clean up request

Jaclyn,

Hope you are doing well. I am reaching out to you in regards to #2511 UC Folder/ Task List Bug and Clean up request. I was looking over this again and although I may have the want and gist of the request, I would like to make sure I am right and schedule a quick user research session with me.
The user research session will provide me a confirmation of what is wanted/needed, a better understanding of the UC folder and how it is used, as well as, what the folder/user/Department user go through to create the folder and get items on/off the task list.

So far, I can see resolving this permanently by having the procedure/process remove the folder from the User's task list by adding the End date automatically when the folder is set to a particular status on the Initial Distribution Process and/or the Permit Completeness Check Process or Make it to where the UC folder is not dependent on an End date to come off of the Task list once it is put in a particular status.

This should be a quick session as I already have Isaiah’s list. I also wanted to see if we could go over the status attempt results description/verbiage, just to see if they are a  correct description for what it should say. 

Please let me know a good time you are available to schedule this.

Respectfully,

Nadin A. Nader 
IT Business Systems Analyst
Data and Technology Services 
Austin Transportation Department


Waiting on Jaclyn's response

**Comment by Nadin-Nader** [comment]
@TracyLinder moving to backlog

**Comment by Nadin-Nader** [comment]
Spoke to Jaclyn in our meeting for this  today. We went over the workflow from when the User submits an application Until the Folder comes off of the Task list 
**(see recording: [https://web.microsoftstream.com/video/3e891468-ef79-4b1e-9a42-c951e45bb97b](url)

Jaclyn wants to mirror the automatic date entered in the Application Review Process to the Initial Distribution Process (When a start date and time are entered as a result on the process tab, the End Date and time mirror the Start date and time.) It shouldn't leave the task list until the user reviews and accepts it but the user shouldn't have to enter the End date and time manually. They only need to review the two processes on the task list, save and it should disappear.

Initial distribution Process is the last step the user has to accept before the task is off of their task list.

Permit completion End Date is automatically entered when a percentage is entered as an attempt in the process Tab and saved.
Application Review End Date is automatically entered when it is accepted (an Accepted" attempt result is entered) in the process tab.
Initial Distribution stays open even after an End Date and "Accepted" result/attempt is entered in the Process tab. Once refreshed, the End Date and Time entered are shown in the Started Date and Time of the process and keeps it open.

@TracyLinder they do the Application Review Process when they do the Web Acceptance Process, are both needed or can they be consolidated or can the Application Review be "Accepted" when the Web Acceptance Process is done?

When all is accepted and the user pulls up the task list; two Items are added for the one application. One is the Permit Completeness Check and the other is the Initial Distribution on Task List. Happens every time.

##They would like to have the Initial Distribution Process End Date on their task list to automatically Populate and mirror the End Date and Time entered from the Process tab.

**Comment by TracyLinder** [comment]
@Nadin-Nader Great Work!

Re: Your question above for me. The web application process is mandatory for all applications coming in through portal. If the UC team is OK with getting rid of the Application Review process, that's the one that should be cut.

**Comment by TracyLinder** [comment]
@Nadin-Nader can you please update this issue's description to better define what the user story is. Eventually I'd like to see acceptance criteria too. Want any help with this one?

**Comment by Nadin-Nader** [comment]
@licarir @TracyLinder 
Sent Email to Isaiah asking for a date range and his response was:

> Dear Nadin
> 
> The 1st June 2016 to the 30th June 2019 works for me.
> 
> Thank you
> 
> Isaiah Lewallen
> Engineering Associate C
> Austin Transportation Department
> Right-of-way Management Division
> 
> 512-974-1479
> 1501 Toomey Rd
> Austin, Texas 78704
> 
> ________________________________________
> From: Nader, Nadin <Nadin.Nader@austintexas.gov>
> Sent: Thursday, May 21, 2020 10:39 AM
> To: Lewallen, Isaiah <Isaiah.Lewallen@austintexas.gov>
> Cc: Lozano, Jaclyn <Jaclyn.Lozano@austintexas.gov>
> Subject: UC Task List Clean Up (#1673) 
>  
> Isaiah,
>  
> Hope you are doing well, I am reaching out to you regarding the UC Task List clean up for your AMANDA profile. We have a new team dedicated to us from CTM AMANDA  to help us with requests and they are working on this one. 
>  
> When I presented them with the request and the spreadsheet I have (attached to this email), they asked if it was these specific ones or if they can use a date range? If you would like them to use a date range, can you please provide me that. If not, they will clean up only the ones on the list and it will also take a bit longer to do.
>  
> Please let me know as soon as you can. Once they get that information, they will be able to start working on this.
>  
> Respectfully,
>  
> Nadin A. Nader 
> IT Business Systems Analyst
> Data and Technology Services 
> Austin Transportation Department
>

**Comment by Nadin-Nader** [comment]
@TracyLinder Updated the description. Let me know what i missed or if that is ok.

**Comment by TracyLinder** [comment]
> @TracyLinder Updated the description. Let me know what i missed or if that is ok.

Looks great, thanks @Nadin-Nader !

**Comment by licarir** [comment]
@Nadin-Nader On your comment on May 18th, there is a link to a video. I can't seem to open it. There is a lot of back and forth here. I need to know what the immediate steps are before and after they expect to have a date filled in on the Initial Distribution process. Are they on the Initial Distribution and they add and save a specific attempt and then it closes all of the other processes but needs to add an end date to itself? Are there other tasks being left open that need to be addressed as well and do those ones get touched individually, or are they all closed by one main process? If you could get me the step by step that looks like the following, that will help me finish this task.
1. Go to Initial Distribution process on UC folder.
2. Add Complete attempt and click save
3. This fires off code which closes out Initial Distribution as well as Permit Completeness Check, Right-of-Way Review, etc.
4. Other processes have end dates but Initial Distribution does not.

Let me know if you want to set up a meeting to clarify and walk me through this.
Thank you.

**Comment by licarir** [comment]
@Nadin-Nader are you the proper point of contact for questions such as the above?

**Comment by licarir** [comment]
@TracyLinder @Nadin-Nader 

I found a clause in the process attempt code for COMPLETE attempts on the Initial Distribution process that make it not updated that folder process when complete attempt is used. Do you know why this would be and if we should make this not happen for only certain Folder > Sub & work > Process scenarios? For instance, we update all of the dates and close the process  when a UC folder of a certain sub & work are being used. 

   UPDATE folderprocess fp 
      SET fp.scheduleenddate = ln_infodate, 
          fp.ScheduleDate = sysdate
    WHERE fp.processcode **not in (82005,51095,51070,51065,70000,61060)** 
      AND fp.folderrsn = argfolderrsn 
      AND fp.statuscode = 1;

**Comment by TracyLinder** [comment]
**Status:** Robert will continue to develop this in sprint 28

**Comment by licarir** [comment]
@Nadin-Nader please peer review commit for this change so that I can request it be moved to Dev.
http://coaamandadevops.austintexas.gov/R3:b0569efddf0abe7a870bf97bc672aaf2e5382d4d
Thank you.

**Comment by licarir** [comment]
@Nadin-Nader This is on Dev ready for you to review. Thank you for auditing these. Initial Distribution needs to be removed from task list when Complete is chosen on the UC folder. (Dates need to be filled in automatically then you can review task list to ensure it doesn't show anymore.

**Comment by licarir** [comment]
Meeting requested for 7/15/2020 morning to review on Dev

**Comment by TracyLinder** [comment]
### Meeting Recap:
**Attendees:** Robert, Nate, Kaarthy, Jaclyn L.
- Robert demonstrated the work/features in DEV. 
- Jaclyn approved the features for Test
- Next steps: Move to Test, Customer UAT

**Comment by TracyLinder** [comment]
There may be a few out-of-scope elements on this issue that should be developed separately.

**Comment by licarir** [comment]
@TracyLinder @Nadin-Nader I broke this down and only fixed Initial Distribution 51070. The other processes are different and I can update them in future iterations. If you see this bug on other processes, listed at the top of this card, then please create new issues for each process to make the fields filled in which will 'close' the task and remove it from the task list without having to manually enter it.

Right-of-Way Admin 50075
Permit Completeness Check 51065
Done --- Initial Distribution process code 51070
Permit Case Manager Review 51095
Right-of-Way Review 51295
Adjust Review Due Dates 61060

**Comment by Nadin-Nader** [comment]
DEVOPS Card to move to TEST #3405

**Comment by licarir** [comment]
@Nadin-Nader @TracyLinder SCTASK0080452 sent to DevOps CM team board for Deploy to TEST.

**Comment by licarir** [comment]
@Nadin-Nader @TracyLinder UC Folder Task list bug fix was deployed to TEST for UAT on 7/17. Please let me know when you are able to UAT. Thank you.

**Comment by licarir** [comment]
What we need to complete this request:
Process name and code if available (IE: 50075 Right Of Way Admin)
Attempt result that will remove task from list (IE 12345 Completed)
Fields to be updated, Status, Assigned, Signoff, Start, End, scheduled start and end Date
Folder Type for given process (UC, RW, EX, Etc.)
Does code on attempt need to run whether process is closed or not. Does attempt close or affect other processes

Sample specification for Task List removal on added attempt for a given process:
50075 RIGHT OF way admin ON UC folders, WHEN attempt withdrawn IS added, SET status closed, assigned USER whoever, END DATE = sysdate, scheduled END DATE, BEGIN DATE, scheduled DATE. If date exists, keep date in field. Start date is already non-null, then keep date in field, otherwise sysdate.

**Comment by licarir** [comment]
@TracyLinder @Nadin-Nader Please UAT this on Test so that we can get this promoted to PROD.

**Comment by TracyLinder** [comment]
**Status:** Nate will facilitate testing w/ stakeholders during sprint 30. Updating milestone to Sprint 30.

**Comment by Nadin-Nader** [comment]
RELATED TO #3498

**Comment by Nadin-Nader** [comment]
**Status:**
Meeting Scheduled for Monday, August 3, 2020

**Comment by Nadin-Nader** [comment]
In Yesterday's meeting with Tracy, Kaarthy, Robert, Isaiah, Jaclyn and myself we discussed the UC Task List bug doc (document attached) and gather information on what is needed to resolve the issue.https://app.zenhub.com/files/140626918/36c802d5-6599-4227-8526-1cc5ae88067c/download

New Info from template:
Attempt result that will remove task from list (IE 12345 Completed)
All valid attempt result should remove it from the task list.
Right-of-Way Admin 50075 –
Permit Case Manager Review 51095 –
Initial Distribution process code 51070 -
Right-of-Way Review 51295 -
Adjust Review Due Dates 61060 –

Permit Completeness Check 51065 - Leave it for Last as it is used multiple times and leave unassigned until users pull it in.

Fields to be updated, Status, Assigned, Signoff, Start, End, Scheduled Start and End Date

What Folder Type for given process (UC, RW, EX, Etc.)
Only for UC Folder

Does the code on the attempt need to run whether process is closed or not?
Permit Completeness check gets ran more than once so this one needs to run either way

Does attempt close or affect other processes?
Initial distribution process, when completed, sends the folder to a user and shows up on the users task list

**Comment by Nadin-Nader** [comment]
@licarir i have UATd this on TEST and it looks good to me.

**Comment by licarir** [comment]
This has made it to PROD

**Comment by TracyLinder** [comment]
Nate, will you please inform the stakeholders and then close this one out?

**Comment by Nadin-Nader** [comment]
Good Morning,

Hope you two are doing well. I am reaching out to you to inform you that 2511 UC Folder Task List bug is not on PROD. I will go ahead and close it out. If you have any issues or concerns please feel free to reach out.

Respectfully,

Nadin A. Nader 
IT Business Systems Analyst
Data and Technology Services 
Austin Transportation Department

**Comment by TracyLinder** [comment]
There’s a typo down there. This is definitely in Prod now. 🎉

All the Best,

Tracy Linder (she/her)
Product Manager, Data & Technology Services
Austin Transportation Department
——————————————————————
data.mobility.austin.gov<https://gcc01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdata.mobility.austin.gov%2F&data=02%7C01%7CTracy.Linder%40austintexas.gov%7C7945cc805599404f4caf08d8359631ff%7C5c5e19f6a6ab4b45b1d0be4608a9a67f%7C0%7C0%7C637318266481081651&sdata=2eb6bM92Yns5NEu7hvUCknL3Ao%2BE95ja7YZB2NQtTao%3D&reserved=0>
